pkg/operator/controller/externaldns/controller.go

```diff
@@ -88,16 +93,16 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 
 	if _, _, err := r.ensureExternalDNSNamespace(ctx, r.config.Namespace); err != nil {
-		// Return if the externalDNS namespace cannot be created since
-		// resource creation in a namespace that does not exist will fail.
-		return result, fmt.Errorf("failed to ensure externalDNS namespace: %v", err)
+		return requeueResult, fmt.Errorf("failed to ensure externalDNS namespace: %w", err)
```

Comment:
Specifying a non-zero value result is superfluous when the error value is non-nil.  Specifying `Requeue` or `RequeueAfter` is only really needed if there was no error but you want to terminate reconciliation and start it over from the beginning.

Reply:
I thought of not re-queuing right after the failure (of any secondary resource creation). Some errors would not change after waiting for some time or not but some might disappear. Also, if it's not a fixable problem logging an error with some re-queue time in between would not pollute the logs that fast.

Reply:
I mean, you're returning `fmt.Errorf(...)`, which is a non-nil error value, and controller-runtime will requeue the reconciliation request immediately anyway when it gets that non-nil error value, so using `return result, fmt.Errorf(...)` should have the same effect as using `return requeueResult, fmt.Errorf(...)`.

Reply:
It's a pity it's not possible to specify `RequeueAfter` for error cases. I removed all requeue logic as for the moment it doesn't make much sense for the success case.